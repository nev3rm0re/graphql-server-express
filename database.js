const mysql = require('mysql2/promise');
const pools = {};

const getConnection = async (databaseName = process.env.DEBUG_TARGET_NAME) => {
  if (!pools[databaseName]) {
    pools[databaseName] = mysql.createPool({
      host: 'localhost',
      user: 'root',
      password: '',
      port: 3307,
      database: databaseName,
      waitForConnections: true,
      connectionLimit: 10,
      queueLimit: 0,
    });
  }
  return await pools[databaseName];
};

module.exports = getConnection;
