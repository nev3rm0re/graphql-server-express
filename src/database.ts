import mysql, { RowDataPacket } from 'mysql2/promise';
import Pool from 'mysql2/typings/mysql/lib/Pool';

const pools: { [dbName: string]: mysql.Pool } = {};

const createPool = async (database: string): Promise<mysql.Pool> => {
  return await mysql.createPool({
    host: 'localhost',
    user: 'root',
    password: '',
    database,
    port: 3307,
    waitForConnections: true,
    connectionLimit: 10,
    queueLimit: 0,
  });
};

/**
 * Returns MySQL Connection pool configured for `databaseName`
 */
export const getConnection = async (
  databaseName: string = process.env.DEBUG_TARGET_NAME || '',
): Promise<mysql.Pool> => {
  if (!pools[databaseName]) {
    pools[databaseName] = await createPool(databaseName);
  }
  return pools[databaseName];
};

export const getDatabases = async (like: string = '%'): Promise<string[]> => {
  const pool = await mysql.createPool({
    host: 'localhost',
    user: 'root',
    password: '',
    port: 3307,
    connectionLimit: 10,
  });

  let databaseQuery = 'SHOW DATABASES';
  if (like !== '%') {
    databaseQuery += ` LIKE '${like}'`;
  }

  console.log('Running query', databaseQuery);
  const [result] = await pool.query(databaseQuery);
  const results = Object.values(result).map(
    (textRow: RowDataPacket): string => {
      console.log('textRow', textRow);
      return Object.values(textRow)[0];
    },
  );
  console.log('Results', results);
  return results;
};
