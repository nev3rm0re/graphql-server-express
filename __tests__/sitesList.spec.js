require('isomorphic-fetch');
import { getConnection } from '../src/database.ts';

const queryServer = (query, variables = {}) => {
  return fetch(
    // 'http://104.154.240.101/graphql',
    'http://localhost:4000/graphql',
    {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        query,
        variables,
      }),
    },
  ).then((res) => res.json());
};
describe('Query sites', () => {
  jest.setTimeout(20000);

  test('query missing entities', async () => {
    const result = await queryServer(`query { missingEntities }`, {});
    expect(result.data).toMatchSnapshot();
  });

  test('what the heck mysql2.query returns', async () => {
    const conn = await getConnection('2008ownersclub.co.uk_target');
    try {
      const [result] = await conn.query('SHOW TABLES LIKE "%_migration"');
      const tables = result.map((row) => Object.values(row)[0]);
      console.debug(tables);
    } catch (err) {
      console.error(err);
    }
  });
  test('SITES query returns expected result', async () => {
    expect.assertions = 1;
    const query = `query {
        sites {
          env
          sitename
          migrated
          source
        }
    }`;

    const result = await queryServer(query);
    expect(result.data).toMatchSnapshot();

    console.log('done getting result', result.data);
    return result;
  });

  test('SITES parametrized query returns expected result', async () => {
    expect.assertions = 1;
    const query = `query($requireSource: Boolean) {
      sites(requireSource: $requireSource) {
        env
        sitename
        migrated
        source
        retroactive
      }
    }`;

    const result = await queryServer(query, { requireSource: true });

    expect(result.data).toMatchSnapshot();
  });
});
