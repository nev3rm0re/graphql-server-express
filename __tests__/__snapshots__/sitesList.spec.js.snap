// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query sites SITES query returns expected result 1`] = `
Object {
  "sites": Array [
    Object {
      "env": "1130cc",
      "migrated": null,
      "sitename": "1130cc.com",
    },
    Object {
      "env": "1911talk",
      "migrated": null,
      "sitename": "1911talk.com",
    },
    Object {
      "env": "2008ownersclub",
      "migrated": null,
      "sitename": "2008ownersclub.co.uk",
    },
    Object {
      "env": "68forums",
      "migrated": null,
      "sitename": "68forums.com",
    },
    Object {
      "env": "allfordmustangs",
      "migrated": null,
      "sitename": "allfordmustangs.com",
    },
    Object {
      "env": "artistforum",
      "migrated": null,
      "sitename": "artistforum.com",
    },
    Object {
      "env": "astrosafari",
      "migrated": null,
      "sitename": "astrosafari.com",
    },
    Object {
      "env": "audiq3forum",
      "migrated": null,
      "sitename": "audiq3forum.com",
    },
    Object {
      "env": "basketballforum",
      "migrated": null,
      "sitename": "basketballforum.com",
    },
    Object {
      "env": "blowgunforum",
      "migrated": null,
      "sitename": "blowgunforum.com",
    },
    Object {
      "env": "cadillacatsforum",
      "migrated": null,
      "sitename": "cadillacatsforum.com",
    },
    Object {
      "env": "camaros",
      "migrated": null,
      "sitename": "camaros.net",
    },
    Object {
      "env": "canamforum",
      "migrated": null,
      "sitename": "can-amforum.com",
    },
    Object {
      "env": "chevroletcruze",
      "migrated": null,
      "sitename": "chevroletcruze.net",
    },
    Object {
      "env": "chevybolt",
      "migrated": null,
      "sitename": "chevybolt.org",
    },
    Object {
      "env": "cichlidforum",
      "migrated": null,
      "sitename": "cichlid-forum.com",
    },
    Object {
      "env": "clubtouareg",
      "migrated": null,
      "sitename": "clubtouareg.com",
    },
    Object {
      "env": "clubtread",
      "migrated": null,
      "sitename": "clubtread.com",
    },
    Object {
      "env": "cruzetalk",
      "migrated": null,
      "sitename": "cruzetalk.com",
    },
    Object {
      "env": "crxcommunity",
      "migrated": null,
      "sitename": "crxcommunity.com",
    },
    Object {
      "env": "cumminsforum",
      "migrated": null,
      "sitename": "cumminsforum.com",
    },
    Object {
      "env": "custommagnums",
      "migrated": null,
      "sitename": "custommagnums.com",
    },
    Object {
      "env": "defenderforums",
      "migrated": null,
      "sitename": "defenderforums.net",
    },
    Object {
      "env": "digitalhome",
      "migrated": null,
      "sitename": "digitalhome.ca",
    },
    Object {
      "env": "dogfoodchat",
      "migrated": null,
      "sitename": "dogfoodchat.com",
    },
    Object {
      "env": "dseries",
      "migrated": null,
      "sitename": "d-series.org",
    },
    Object {
      "env": "evotuners",
      "migrated": null,
      "sitename": "evotuners.net",
    },
    Object {
      "env": "focusrsclub",
      "migrated": null,
      "sitename": "focusrsclub.com",
    },
    Object {
      "env": "focusst",
      "migrated": null,
      "sitename": "focusst.org",
    },
    Object {
      "env": "fordescape",
      "migrated": null,
      "sitename": "fordescape.org",
    },
    Object {
      "env": "fordtough",
      "migrated": null,
      "sitename": "fordtough.ca",
    },
    Object {
      "env": "fullsizebronco",
      "migrated": null,
      "sitename": "fullsizebronco.com",
    },
    Object {
      "env": "fz07oc",
      "migrated": null,
      "sitename": "fz07oc.com",
    },
    Object {
      "env": "g5club",
      "migrated": null,
      "sitename": "g5club.net",
    },
    Object {
      "env": "g8board",
      "migrated": null,
      "sitename": "g8board.com",
    },
    Object {
      "env": "gacforums",
      "migrated": null,
      "sitename": "gacforums.com",
    },
    Object {
      "env": "gminsidenews",
      "migrated": null,
      "sitename": "gminsidenews.com",
    },
    Object {
      "env": "goosehuntingchat",
      "migrated": null,
      "sitename": "goosehuntingchat.com",
    },
    Object {
      "env": "gt350",
      "migrated": null,
      "sitename": "gt350.org",
    },
    Object {
      "env": "halloweenforum",
      "migrated": null,
      "sitename": "halloweenforum.com",
    },
    Object {
      "env": "havaneseforum",
      "migrated": null,
      "sitename": "havaneseforum.com",
    },
    Object {
      "env": "haytalk",
      "migrated": null,
      "sitename": "haytalk.com",
    },
    Object {
      "env": "hdlivewireforum",
      "migrated": null,
      "sitename": "hdlivewireforum.com",
    },
    Object {
      "env": "hellcat",
      "migrated": null,
      "sitename": "hellcat.org",
    },
    Object {
      "env": "hobbytalk",
      "migrated": null,
      "sitename": "hobbytalk.com",
    },
    Object {
      "env": "horseforum",
      "migrated": null,
      "sitename": "horseforum.com",
    },
    Object {
      "env": "hotrodders",
      "migrated": null,
      "sitename": "hotrodders.com",
    },
    Object {
      "env": "hrvforum",
      "migrated": null,
      "sitename": "hrvforum.com",
    },
    Object {
      "env": "hyundaicoupeclub",
      "migrated": null,
      "sitename": "hyundaicoupeclub.co.uk",
    },
    Object {
      "env": "icefishingchat",
      "migrated": null,
      "sitename": "icefishingchat.com",
    },
    Object {
      "env": "ilxforums",
      "migrated": null,
      "sitename": "ilxforums.com",
    },
    Object {
      "env": "indianmotorcycles",
      "migrated": null,
      "sitename": "indianmotorcycles.net",
    },
    Object {
      "env": "jeepforum",
      "migrated": null,
      "sitename": "jeepforum.com",
    },
    Object {
      "env": "jenkins",
      "migrated": null,
      "sitename": "unittest",
    },
    Object {
      "env": "kiaforums",
      "migrated": null,
      "sitename": "kia-forums.com",
    },
    Object {
      "env": "liveartist",
      "migrated": null,
      "sitename": "artistforum.com",
    },
    Object {
      "env": "maseratilife",
      "migrated": null,
      "sitename": "maseratilife.com",
    },
    Object {
      "env": "masterreg",
      "migrated": null,
      "sitename": "masterreg.com",
    },
    Object {
      "env": "minitorque",
      "migrated": null,
      "sitename": "minitorque.com",
    },
    Object {
      "env": "modeltrainforum",
      "migrated": null,
      "sitename": "modeltrainforum.com",
    },
    Object {
      "env": "mondeostoc",
      "migrated": null,
      "sitename": "mondeostoc.com",
    },
    Object {
      "env": "motorgeek",
      "migrated": null,
      "sitename": "motorgeek.com",
    },
    Object {
      "env": "my",
      "migrated": null,
      "sitename": "my.is",
    },
    Object {
      "env": "myfastgti",
      "migrated": null,
      "sitename": "myfastgti.com",
    },
    Object {
      "env": "newninja",
      "migrated": null,
      "sitename": "newninja.com",
    },
    Object {
      "env": "nissanclub",
      "migrated": null,
      "sitename": "nissanclub.com",
    },
    Object {
      "env": "nokiausers",
      "migrated": null,
      "sitename": "nokiausers.net",
    },
    Object {
      "env": "olddefender",
      "migrated": null,
      "sitename": "defenderforums.net",
    },
    Object {
      "env": "paintballforum",
      "migrated": null,
      "sitename": "paintballforum.com",
    },
    Object {
      "env": "palisadeforum",
      "migrated": null,
      "sitename": "palisadeforum.com",
    },
    Object {
      "env": "piloteers",
      "migrated": null,
      "sitename": "piloteers.org",
    },
    Object {
      "env": "piranhafury",
      "migrated": null,
      "sitename": "piranha-fury.com",
    },
    Object {
      "env": "qa",
      "migrated": null,
      "sitename": "jeepforum.com",
    },
    Object {
      "env": "r8talk",
      "migrated": null,
      "sitename": "r8talk.com",
    },
    Object {
      "env": "ratforum",
      "migrated": null,
      "sitename": "ratforum.com",
    },
    Object {
      "env": "rav4world",
      "migrated": null,
      "sitename": "rav4world.com",
    },
    Object {
      "env": "scirocconet",
      "migrated": null,
      "sitename": "scirocconet.co.uk",
    },
    Object {
      "env": "scramblerforum",
      "migrated": null,
      "sitename": "scramblerforum.com",
    },
    Object {
      "env": "seatforum",
      "migrated": null,
      "sitename": "seat-forum.co.uk",
    },
    Object {
      "env": "shotgunworld",
      "migrated": null,
      "sitename": "shotgunworld.com",
    },
    Object {
      "env": "siennachat",
      "migrated": null,
      "sitename": "siennachat.com",
    },
    Object {
      "env": "sitename",
      "migrated": null,
      "sitename": "sitename.com",
    },
    Object {
      "env": "slingshotforums",
      "migrated": null,
      "sitename": "slingshotforums.com",
    },
    Object {
      "env": "speakev",
      "migrated": null,
      "sitename": "speakev.com",
    },
    Object {
      "env": "srtforums",
      "migrated": null,
      "sitename": "srtforums.com",
    },
    Object {
      "env": "ssrfanatic",
      "migrated": null,
      "sitename": "ssrfanatic.com",
    },
    Object {
      "env": "stingrayforums",
      "migrated": null,
      "sitename": "stingrayforums.com",
    },
    Object {
      "env": "subaruforester",
      "migrated": null,
      "sitename": "subaruforester.org",
    },
    Object {
      "env": "taurusclub",
      "migrated": null,
      "sitename": "taurusclub.com",
    },
    Object {
      "env": "tbssowners",
      "migrated": null,
      "sitename": "tbssowners.com",
    },
    Object {
      "env": "tccoa",
      "migrated": null,
      "sitename": "tccoa.com",
    },
    Object {
      "env": "test",
      "migrated": null,
      "sitename": "unittest",
    },
    Object {
      "env": "thenewx",
      "migrated": null,
      "sitename": "thenewx.org",
    },
    Object {
      "env": "theprofilebrotherhood",
      "migrated": null,
      "sitename": "theprofilebrotherhood.com",
    },
    Object {
      "env": "tidalfish",
      "migrated": null,
      "sitename": "tidalfish.com",
    },
    Object {
      "env": "tractorforum",
      "migrated": null,
      "sitename": "tractorforum.com",
    },
    Object {
      "env": "tradtalk",
      "migrated": null,
      "sitename": "tradtalk.com",
    },
    Object {
      "env": "trailvoy",
      "migrated": null,
      "sitename": "trailvoy.com",
    },
    Object {
      "env": "tundratalk",
      "migrated": null,
      "sitename": "tundratalk.net",
    },
    Object {
      "env": "v6z24",
      "migrated": null,
      "sitename": "v6z24.com",
    },
    Object {
      "env": "vtwinforum",
      "migrated": null,
      "sitename": "v-twinforum.com",
    },
  ],
}
`;
